src/sempy_labs/_notebooks.py

```
@@ -284,3 +298,141 @@ def update_notebook_definition(
+            notebook_name = n["Notebook Name"]
+            definition = _base_api(
+                request=f"v1/workspaces/{w_id}/notebooks/{notebook_id}/getDefinition",
+                method="post",
+                client="fabric_sp",
+                status_codes=None,
+                lro_return_json=True,
+            )
+            for part in definition.get("definition").get("parts"):
+                payload = _decode_b64(part["payload"])
+                if part["path"] == "notebook-content.py":
+                    if search_string in payload:
```

Ruff (0.12.2)

Use a single `if` statement instead of nested `if` statements

(SIM102)